.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,56 @@
+            "openedx/core/djangoapps/geoinfo/ openedx/core/djangoapps/header_control/ openedx/core/djangoapps/heartbeat/ openedx/core/djangoapps/lang_pref/ openedx/core/djangoapps/models/ openedx/core/djangoapps/monkey_patch/ openedx/core/djangoapps/oauth_dispatch/ openedx/core/djangoapps/olx_rest_api/ openedx/core/djangoapps/password_policy/ openedx/core/djangoapps/plugin_api/ openedx/core/djangoapps/plugins/ openedx/core/djangoapps/profile_images/ openedx/core/djangoapps/programs/ openedx/core/djangoapps/safe_sessions/ openedx/core/djangoapps/schedules/ openedx/core/djangoapps/self_paced/ openedx/core/djangoapps/service_status/ openedx/core/djangoapps/session_inactivity_timeout/ openedx/core/djangoapps/signals/ openedx/core/djangoapps/site_configuration/ openedx/core/djangoapps/system_wide_roles/ openedx/core/djangoapps/theming/ openedx/core/djangoapps/user_api/ openedx/core/djangoapps/user_authn/ openedx/core/djangoapps/util/ openedx/core/djangoapps/verified_track_content/ openedx/core/djangoapps/video_config/ openedx/core/djangoapps/video_pipeline/ openedx/core/djangoapps/waffle_utils/ openedx/core/djangoapps/xblock/ openedx/core/djangoapps/xmodule_django/ openedx/core/djangoapps/zendesk_proxy/ openedx/core/djangolib/ openedx/core/lib/ openedx/core/tests/ openedx/core/tests/ openedx/features/ openedx/testing/ openedx/tests/",
+            "cms/djangoapps/api/ cms/djangoapps/cms_user_tasks/ cms/djangoapps/course_creators/ cms/djangoapps/export_course_metadata/ cms/djangoapps/maintenance/ cms/djangoapps/models/ cms/djangoapps/pipeline_js/ cms/djangoapps/xblock_config/ cms/envs/ cms/lib/",
+            "cms/djangoapps/contentstore/",
+            "common/djangoapps/",
+            "common/lib/",
+        ]
+
+
+    name: python ${{ matrix.python-version }},django ${{ matrix.django-version }} ${{ matrix.test_module }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: start mongodb service
+        run: |
+          sudo /etc/init.d/mongodb start
+
+      - name: set module name
+        run: |
+          echo "module_name=$(echo '${{ matrix.test_module }}' | awk -F '/' '{print $1}')" >> $GITHUB_ENV
+
+      - name: set settings path
+        run: |
+          echo "settings_path=$(if [ '${{ env.module_name }}' = 'cms' ]; then echo 'cms.envs.test'; else echo 'lms.envs.test' ; fi)" >> $GITHUB_ENV
+
+      - name: install requirements
+        run: |
+          sudo pip install -r requirements/pip.txt
+          sudo pip install -r requirements/edx/testing.txt
+          sudo pip install "django~=${{ matrix.django-version }}.0"
+
+      - name: run tests
+        run: |
+          python -Wd -m pytest -p no:randomly --ds=${{ env.settings_path }} ${{ matrix.test_module }}
```

Comment:
1. Were tests randomly ordered in Jenkins? Is that something we are losing and need to lose? 
2. It would be great if there were an ADR that captured the decision to move and listed all of the consequences.

Reply:
@iamsobanjaved @mraarif: On a related note, it looks like you may have run into an order dependent test issue. Whether as part of a separate ticket or not, there is a script that could help find the fewest number of tests needed to recreate an order dependent problem, but it relied on Jenkins input to get started, so it would need some adjustments to work.

Related to my earlier comment, are we ok with a fixed order and having order dependent issues creep in over time while not testing randomly?

Reply:
Personally, I'd be OK with using a fixed order if means we can get off of Jenkins sooner, as long as we then have a high priority to use automated test selection and randomized sharding.

Reply:
1. in Jenkins we're also using this flag `no:randomly` , without this some tests start failing, we should ideally don't have to specify this flag
2. I'm not that aware of how an ADR works but I'd be documenting the reasons for these choices/decisions

yes, we've run into a test failure where reordering a Django app under `openedx` (`openedx/core/djangoapps/cors_csrf`) in one of the shards fixed a test that was failing

for other modules (e.g `lms`, `cms`), we've listed the apps in order in which they appear in our code

Reply:
yes there's a Github workflow called `Ensure unit tests running` (https://github.com/edx/edx-platform/actions/workflows/unit-tests-check.yml) that runs `pytest --collect-only` for that purpose, but with this new check (https://github.com/edx/edx-platform/pull/29291) that verifies the exact number of tests collected, we would no longer need `Ensure unit tests running`, I've also noticed that check (`Ensure unit tests running`) is only collecting 26 tests and we've 23596 unit tests in our codebase as per this check: https://github.com/edx/edx-platform/runs/4163353998?check_suite_focus=true

Reply:
1. Regarding test randomization, that is definitely not a blocker for this PR and work, but it sounds like we should iterate to achieve that where possible.
2. An ADR is just a standard way to document your decisions in the code base. It sounds like you may have plans for documenting these decisions, and it is nice when the decisions aren't buried inside JIRA comments, etc., or lost altogether.
3. Glad you have what may be an improved check that all tests are running.

Reply:
@jmbowman , I've tested `no:randomly` with `cms` and `common`, tests for all the other work fine except for one of the `openedx` shard that fails with around 4 to 20 test failures randomly, we'll definitely create a ticket for this to look into and fix